lib/app.js

```diff
@@ -122,48 +124,41 @@ module.exports = function(tilelive, options) {
     sourceURIs[scale] = retinaURI;
   });
 
-  app.get(tilePattern, function(req, res, next) {
-    var z = req.params.z | 0,
-        x = req.params.x | 0,
-        y = req.params.y | 0,
-        scale = (req.params.scale || "@1x").slice(1, 2),
-        sourceURI = sourceURIs[scale],
+  var getTile = function(z, x, y, scale, format, callback) {
+    var sourceURI = sourceURIs[scale],
         params = {
           tile: {
             zoom: z,
             x: x,
             y: y,
-            format: req.params.format,
+            format: format,
             retina: scale > 1,
             scale: scale
           }
         };
 
-
     return tilelive.load(sourceURI, function(err, source) {
       if (err) {
-        return next(err);
+        return callback(err);
       }
 
       return tessera.getInfo(source, function(err, info) {
         if (err) {
-          return next(err);
+          return callback(err);
         }
 
         // validate format / extension
         var ext = getExtension(info.format);
 
-        if (ext !== req.params.format) {
-          debug("Invalid format '%s', expected '%s'", req.params.format, ext);
-          res.set(populateHeaders({}, params, { 404: true, invalidFormat: true }));
-          return res.status(404).end();
+        if (ext !== format) {
+          debug("Invalid format '%s', expected '%s'", format, ext);
+          return callback(404, 'Invalid format', populateHeaders({}, params, { 404: true, invalidFormat: true }));
```

Comment:
This (and the subsequent callbacks) should pass an `Error` instance as the first argument for consistency.